tests/test_bootstrap.py

```diff
@@ -233,3 +191,169 @@ def gen_domain_setup(domain):
+        yield enable_logging
+
+
+@pytest.fixture
+def mock_mount_local_lib_path():
+    """Mock enable logging."""
+    with patch(
+        "homeassistant.bootstrap.async_mount_local_lib_path"
+    ) as mount_local_lib_path:
+        yield mount_local_lib_path
+
+
+@pytest.fixture
+def mock_process_ha_config_upgrade():
+    """Mock enable logging."""
+    with patch(
+        "homeassistant.config.process_ha_config_upgrade"
+    ) as process_ha_config_upgrade:
+        yield process_ha_config_upgrade
+
+
+@pytest.fixture
+def mock_ensure_config_exists():
+    """Mock enable logging."""
+    with patch(
+        "homeassistant.config.async_ensure_config_exists", return_value=True
+    ) as ensure_config_exists:
+        yield ensure_config_exists
+
+
+async def test_setup_hass(
+    mock_enable_logging,
+    mock_is_virtual_env,
+    mock_mount_local_lib_path,
+    mock_ensure_config_exists,
+    mock_process_ha_config_upgrade,
+):
+    """Test it works."""
+    verbose = Mock()
+    log_rotate_days = Mock()
+    log_file = Mock()
+    log_no_color = Mock()
+
+    with patch(
+        "homeassistant.config.async_hass_config_yaml", return_value={"browser": {}}
+    ):
+        hass = await bootstrap.async_setup_hass(
+            config_dir=get_test_config_dir(),
+            verbose=verbose,
+            log_rotate_days=log_rotate_days,
+            log_file=log_file,
+            log_no_color=log_no_color,
+            skip_pip=True,
+            safe_mode=False,
+        )
+
+    assert "browser" in hass.config.components
+
+    assert len(mock_enable_logging.mock_calls) == 1
+    assert mock_enable_logging.mock_calls[0][1] == (
+        hass,
+        verbose,
+        log_rotate_days,
+        log_file,
+        log_no_color,
+    )
+    assert len(mock_mount_local_lib_path.mock_calls) == 1
+    assert len(mock_ensure_config_exists.mock_calls) == 1
+    assert len(mock_process_ha_config_upgrade.mock_calls) == 1
+
+
+async def test_setup_hass_invalid_yaml(
+    mock_enable_logging,
+    mock_is_virtual_env,
+    mock_mount_local_lib_path,
+    mock_ensure_config_exists,
+    mock_process_ha_config_upgrade,
+):
+    """Test it works."""
+    with patch(
+        "homeassistant.config.async_hass_config_yaml", side_effect=HomeAssistantError
+    ):
+        hass = await bootstrap.async_setup_hass(
+            config_dir=get_test_config_dir(),
+            verbose=False,
+            log_rotate_days=10,
+            log_file="",
+            log_no_color=False,
+            skip_pip=True,
+            safe_mode=False,
+        )
+
+    assert "safe_mode" in hass.config.components
+    assert len(mock_mount_local_lib_path.mock_calls) == 0
+
+
+async def test_setup_hass_config_dir_nonexistent(
+    mock_enable_logging,
+    mock_is_virtual_env,
+    mock_mount_local_lib_path,
+    mock_ensure_config_exists,
+    mock_process_ha_config_upgrade,
+):
+    """Test it works."""
+    mock_ensure_config_exists.return_value = False
+
+    assert (
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4821145)